rpcs3/Emu/Cell/lv2/sys_net.cpp

```diff
@@ -200,12 +200,11 @@ static sys_net_error get_last_error(bool is_blocking, int native_error = 0)
 	}
 
 #ifdef _WIN32
-	switch (native_error)
 #define ERROR_CASE(error) case WSA ## error: result = SYS_NET_ ## error; name = #error; break;
 #else
-	switch (native_error)
 #define ERROR_CASE(error) case error: result = SYS_NET_ ## error; name = #error; break;
 #endif
+	switch (native_error)
```

Comment:
Why are more error types defined on windows than other plateforms?

Reply:
I was just looking at the potential error return list from recvfrom, and Windows had these defined, whilst Linux did not.
Many of them could possibly transition to being outside the windows only section, since they also apply to Linux/Posix systems.

Reply:
All the errors you put in the #ifdef _WIN32 exist on linux.
PS3 is based on BSD which is 99% if not 100% compatible tcp/ip api wise with linux.

Reply:
Do you see any harm in having them all in here? (i.e. all potential mappings from native->PS3)
It shouldn't change our handling of things, but will align the error codes provided to the PS3 app, so app behaviour might improve for error situations.

Reply:
Apart from EINTR and EFAULT which should be handled separately like elad pointed out it's probably fine.
Ofc remove the #ifdef _WIN32.

Reply:
I understand that we need to do some additional handling (outside of this function) in regards to EINTR and EFAULT, but shouldn't we still be translating these native error codes into PS3ese (within this function)?
Otherwise returning invalid error codes to the PS3 land seems like it could cause confusion.

Reply:
You know what, instead of logging an error use fmt::throw_exception.

Reply:
and dont be stubburn about EFAULT and EINTR, drop them.

Reply:
fmt::throw for the default case.. when an error has slipped past translation right?
fmt::throw on all reported errors doesn't seem like a great plan, sometimes they will be intentional.

I'm not sure I have a 'non-stubborn' mode... ;)
But I'll comment out the EFAULT / EINTR for now.

Reply:
yes I mean use it in the defaut case.
well yes its better to left it commented out.